compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -328,15 +328,19 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
 
   private def addInheritanceDependencies(tree: Template)(using Context): Unit =
     if (tree.parents.nonEmpty) {
-      val depContext =
-        if (tree.symbol.owner.isLocal) LocalDependencyByInheritance
-        else DependencyByInheritance
-      val from = resolveDependencySource
-      tree.parents.foreach { parent =>
-        _dependencies += ClassDependency(from, parent.tpe.classSymbol, depContext)
+      withinDepContext(tree.symbol.owner) { (depContext, from) =>
+        for parent <- tree.parents do
+          _dependencies += ClassDependency(from, parent.tpe.classSymbol, depContext)
       }
     }
 
+  private inline def withinDepContext(cls: Symbol)(inline op: (DependencyContext, Symbol) => Context ?=> Unit)(using Context): Unit =
+    val depContext =
+      if cls.isLocal then LocalDependencyByInheritance
+      else DependencyByInheritance
+    val from = resolveDependencySource
+    op(depContext, from)
```

Comment:
I don't think this abstraction is worth it, especially since it's only used in two places, instead I think just having a method that takes a `cls` as input and return `LocalDependencyByInheritance` or `DependencyByInheritance` would be enough, then at use-site you do `val depContext = depContextOf(cls); val from = resolveDependencySource`

Reply:
done